src/main/java/uk/ac/ebi/ampt2d/commons/accession/persistence/jpa/accession/entities/AccessionedLongEntity.java

```diff
@@ -15,30 +15,24 @@
  * limitations under the License.
  *
  */
+package uk.ac.ebi.ampt2d.commons.accession.persistence.jpa.accession.entities;
 
-package uk.ac.ebi.ampt2d.commons.accession.generators;
+import javax.persistence.Id;
+import javax.persistence.MappedSuperclass;
 
-public interface ModelHashAccession<MODEL, HASH, ACCESSION> {
+@MappedSuperclass
+public abstract class AccessionedLongEntity extends AccessionedEntity<Long> {
 
-    MODEL model();
+    @Id
```

Comment:
Sorry for repeating myself so much. Is allowed for an `@Id` column to be non-unique? or this classes will be used only for the id generation (not for the legacy ids import)?

Reply:
@Id marks on creation that column to be a primary key, which is according to PostgreSQL documentation

'''5.3.4. Primary Keys
Technically, a primary key constraint is simply a combination of a unique constraint and a not-null constraint.'''